parser/type.go

```diff
@@ -179,15 +181,16 @@ func parseNominalTypeRemainder(p *parser, token lexer.Token) (*ast.NominalType,
 		// Skip the dot
 		p.next()
 
-		nestedToken := p.current
+		tok := p.current
+		nestedToken = &tok
```

Comment:
Is it OK to override this on each loop iteration? Is the idea to only keep track of the last?

Reply:
Yeah so this was previously an inline variable. I moved it out of the loop precisely because of that - to have access to the last token outside the loop.

I added a comment around that to the `nestedToken` var, but I can clarify it further a bit.